### PR TITLE
Implement native ValueType Equals and GetHashCode support for pointer fields

### LINKED ISSUE #7471: ValueType.Equals doesn't handle pointer fields
## Steps to Reproduce

1. Run following test case
2. 
3. 

```
using System;
class Program
{
    static void Main()
    {
        var a = new StructWithPointer();
        var b = new StructWithPointer();
        Console.WriteLine(a.Equals(b));
    }
}

public unsafe struct StructWithPointer
{
    public int* p;
}
```

## Current Behavior

False

## Expected Behavior

.NET Outputs:
True

### On which platforms did you notice this

[ ] macOS
[ ] Linux
[ X] Windows

**Version Used**:

```
Mono JIT compiler version 5.11.0 (Visual Studio built mono)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           normal
        SIGSEGV:       normal
        Notification:  Thread + polling
        Architecture:  x86
        Disabled:      none
        Misc:          softdebug
        Interpreter:   yes
        LLVM:          supported, not enabled.
        GC:            sgen (concurrent by default)
```

### PR DESCRIPTION
**Pull request by joncham**
Fixes #7471

### CODE REVIEW

File: mono/metadata/icall.c [hunk 1]
```diff
@@ -1238,6 +1242,15 @@ ves_icall_System_ValueType_Equals (MonoObject *this_obj, MonoObject *that, MonoA
 			if (*(double *) this_field != *(double *) that_field)
 				return FALSE;
 			break;
+		case MONO_TYPE_PTR:
+#ifdef NO_UNALIGNED_ACCESS
+			if (G_UNLIKELY ((intptr_t) this_field & 7 || (intptr_t) that_field & 7))
+				UNALIGNED_COMPARE (gpointer);
+			else
+#endif
```
Comment (vargaz) [inline mono/metadata/icall.c]: It would be better to duplicate the check inside both #ifdef cases.

Comment (vargaz) [reply mono/metadata/icall.c]: i.e.
else
  <check>
#else
  <check>
#endif

Comment (vargaz) [reply mono/metadata/icall.c]: Ie the if (*(gpointer *) this_field != *(gpointer *) that_field)
+				return FALSE;
lines

**Comment by joncham** [comment]
@monojenkins build Linux AArch64

**Pull request merged**